### PR TITLE
Use eslint-config-modular-app for our own project

### PR DESCRIPTION
**Pull request by threepointone**
This PR uses eslint-config-modular-app for our own app. It also takes some of the custom rules we were using and adds them to the config.

This also adds .prettierignore files, for ourselves, and generated projects.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,108 +1,29 @@
-// The ESLint browser environment defines all browser globals as valid,
-// even though most people don't know some of them exist (e.g. `name` or `status`).
-// This is dangerous as it hides accidentally undefined variables.
-// We blacklist the globals that we deem potentially confusing.
-// To use them, explicitly reference them, e.g. `window.name` or `window.status`.
-// See https://github.com/facebook/create-react-app/blob/3c2f2d4/packages/eslint-config-react-app/index.js#L19-L24
-const restrictedGlobals = require('confusing-browser-globals');
+'use strict';
 
+// eslint-disable-next-line  no-unused-vars
```
Comment (LukeSheard) [inline .eslintrc.js:3]: What's up with these vars anyway? Can we add a comment so to why they are here?

File: packages/eslint-config-modular-app/index.js [hunk 2]
```diff
@@ -1,5 +1,42 @@
 'use strict';
 
+// eslint-disable-next-line  no-unused-vars
```
Comment (LukeSheard) [inline packages/eslint-config-modular-app/index.js:3]: See other comment - why are these here?

File: packages/modular-scripts/types/app/.eslintignore [hunk 3]
```diff
@@ -0,0 +1 @@
+/public
```
Comment (LukeSheard) [inline packages/modular-scripts/types/app/.eslintignore:1]: Is it not possible to just ignore these as part of the root eslint config? I think the idea of having scattered files around the codebase is confusing - if we can we should keep configuration in the root of the project so that you can conceptually hold it in your head from one source of truth.

File: packages/modular-views.macro/src/__tests__/.eslintrc.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+'use strict';
```
Comment (LukeSheard) [inline packages/modular-views.macro/src/__tests__/.eslintrc.js:1]: Is it not possible to just ignore these as part of the root eslint config? I think the idea of having scattered files around the codebase is confusing - if we can we should keep configuration in the root of the project so that you can conceptually hold it in your head from one source of truth.

File: packages/modular-scripts/types/app/.prettierignore [hunk 5]
```diff
@@ -0,0 +1 @@
+/public
```
Comment (LukeSheard) [inline packages/modular-scripts/types/app/.prettierignore:1]: Is it not possible to just ignore these as part of the root config? I think the idea of having scattered files around the codebase is confusing - if we can we should keep configuration in the root of the project so that you can conceptually hold it in your head from one source of truth.

File: packages/modular-scripts/types/app/.eslintignore (hunk 3 above)
Comment (threepointone) [reply packages/modular-scripts/types/app/.eslintignore]: done

File: packages/modular-scripts/types/app/.prettierignore (hunk 5 above)
Comment (threepointone) [reply packages/modular-scripts/types/app/.prettierignore]: done

File: packages/modular-views.macro/src/__tests__/.eslintrc.js (hunk 4 above)
Comment (threepointone) [reply packages/modular-views.macro/src/__tests__/.eslintrc.js]: this one I think is worth the exception.

File: packages/eslint-config-modular-app/index.js (hunk 2 above)
Comment (threepointone) [reply packages/eslint-config-modular-app/index.js:7]: added a comment

File: .eslintrc.js (hunk 1 above)
Comment (threepointone) [reply .eslintrc.js:7]: added a comment

**Comment by threepointone** [comment]
Going to land this, it's looking good. We can iterate more on it ofc.
